lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -135,7 +135,151 @@ def merge(self, elt1: Element, elt2: Element) -> Element:
 
         return new_elt
 
+class GreedySectionMerger(ElementMerger):
+    def __init__(self, tokenizer: Tokenizer, max_tokens: int, merge_across_pages: bool = True):
+        self.tokenizer = tokenizer
+        self.max_tokens = max_tokens
+        self.merge_across_pages = merge_across_pages
+
+    def preprocess_element(self, element: Element) -> Element:
+        if element.type == 'Image' and "summary" in element.properties:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.properties['summary'] or ""))
+        else:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.text_representation or ""))
+        return element
+
+    def postprocess_element(self, element: Element) -> Element:
+        del element.data["token_count"]
+        return element
+
+    def should_merge(self, element1: Element, element2: Element) -> bool:
+        # deal with empty elements
+        if "page_number" not in element1.properties or "page_number" not in element2.properties or element1.type == None or element2.type == None:
```

Comment:
what are cases where an elements type is none? Is that an invalid state/should we fail?

Reply:
Some of the previous unit tests had empty elements. I agree that they should not be there, in principle.